### ISSUE TITLE
model.py的实现问题

### ISSUE THREAD

**Issue by yajnun2002** [idejie/mml-assignment3#23]
请问train_forward函数里为什么要res = torch.softmax(res, dim=2)？
 nn.CrossEntropyLoss()的预测值是否应当是未经过 softmax 的原始 logits?

**Comment by yajnun2002** [comment]
还有Mapping网络的实现，为何要用nn.TransformerEncoder，是否存在时间维度？能正确处理形状为（Batch_size,Embedding）的img_embedded输入吗？

**Comment by idejie** [comment]
我猜你是用的VIT中对应的CLS_TOKEN对应的embedding或者pooled_output，他的维度应该是1x d, 不用Transformer

**Comment by idejie** [comment]
> 请问train_forward函数里为什么要res = torch.softmax(res, dim=2)？ nn.CrossEntropyLoss()的预测值是否应当是未经过 softmax 的原始 logits?

如果是新的pytorch(如2.x)实现CrossEntropyLoss中会自动计算softmax，因此不需要。

> 还有Mapping网络的实现，为何要用nn.TransformerEncoder，是否存在时间维度？能正确处理形状为（Batch_size,Embedding）的img_embedded输入吗？
我猜你是用的VIT中对应的CLS_TOKEN对应的embedding或者pooled_output，他的维度应该是1x d, 不用Transformer
提示：如果是pool前的呢？
